include/kernels/ADMatReactionFlexible.h

```diff
@@ -0,0 +1,44 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADKernel.h"
+
+/**
+ * This kernel adds to the residual a contribution of \f$ - coeff*L*vs \f$ where
+ * coeff is a simple coefficient, \f$ L \f$ is a material property and \f$ vs \f$
+ * is the multiplication of several variables (nonlinear or coupled).
+ */
+class ADMatReactionFlexible : public ADKernel
+{
+public:
+  static InputParameters validParams();
+
+  ADMatReactionFlexible(const InputParameters & parameters);
+
+protected:
+  virtual ADReal computeQpResidual();
+
+  /**
+   * Kernel variable (can be nonlinear or coupled variable)
+   * (For constrained Allen-Cahn problems, v = lambda
+   * where lambda is the Lagrange multiplier)
+   */
+  const unsigned int _num_vs;
+  const std::vector<const VariableValue *> _vs;
+  // std::vector<const ADVariableValue &> _vs;
+  // const ADVariableValue & _v;
+
+  /// Reaction rate
+  const ADMaterialProperty<Real> & _mob;
+
+  /// An optional input-file supplied coefficient
```

Comment:
What does the coefficient do? Would be nice to state that in the doxygen string

Reply:
This coefficient can be used to add the stoichiometry of a reaction without creating many reaction rates. When you have several species, you can have the same reaction rate for different reactions, but different coefficients. For example:

`X -> 2Y` with reaction rate `K` would use `coeff = 2` for Y, but `coeff = -1` for X. 

I have added a description in the documentation and in the .C file.

Reply:
Thanks. That's very helpful. But I still think it's helpful to have the description in the doxygen in the header as well. This is for different audiences. The documentation in the parameter string is very helpful for a user. The documentation in the header is very helpful for a developer (although admittedly they could go look at the parameter string in the .C file) who quickly peruses the class definition in order to get an understanding of the class. What I usually do is do copy-paste between what I put in the parameter string and what I put in the header doxygen.